tests/lib/EventDispatcher/EventListener/ContentSuggestionSubscriberTest.php

```diff
@@ -8,84 +8,230 @@
         );
     }
 
-    public function testOnContentSuggestion(): void
+    public function testOnContentSuggestionWithLegacyEngineAndScoring(): void
+    {
+        $this->configProviderMock
+            ->method('getRepositoryConfig')
+            ->willReturn(['search' => ['engine' => 'legacy']]);
+        $this->searchServiceSupportsScoring = true;
+        $this->performTestOnContentSuggestion();
+    }
+
+    public function testOnContentSuggestionWithNonLegacyEngineAndScoring(): void
+    {
+        $this->configProviderMock
+            ->method('getRepositoryConfig')
+            ->willReturn(['search' => ['engine' => 'not_legacy']]);
+        $this->searchServiceSupportsScoring = true;
+        $this->performTestOnContentSuggestion();
+    }
+
+    public function testOnContentSuggestionWithLegacyEngineWithoutScoring(): void
+    {
+        $this->configProviderMock
+            ->method('getRepositoryConfig')
+            ->willReturn(['search' => ['engine' => 'legacy']]);
+        $this->searchServiceSupportsScoring = false;
+        $this->performTestOnContentSuggestion();
+    }
+
+    public function testOnContentSuggestionWithNonLegacyEngineWithoutScoring(): void
+    {
+        $this->configProviderMock
+            ->method('getRepositoryConfig')
+            ->willReturn(['search' => ['engine' => 'not_legacy']]);
+        $this->searchServiceSupportsScoring = false;
+        $this->performTestOnContentSuggestion();
+    }
+
+    public function testOnContentSuggestionWithException(): void
+    {
+        $this->configProviderMock
+            ->method('getRepositoryConfig')
+            ->willReturn(['search' => ['engine' => 'legacy']]);
+        $this->searchServiceSupportsScoring = true;
+
+        $query = new SuggestionQuery('test', 10, 'eng-GB');
+
+        $searchService = $this->getSearchServiceMockWithException();
+        $mapper = $this->getSearchHitToContentSuggestionMapperMock();
+        $sortingDefinitionRegistry = $this->getSortingDefinitionRegistryMock();
+
+        $subscriber = new ContentSuggestionSubscriber($searchService, $mapper, $sortingDefinitionRegistry);
+        $subscriber->setLogger($this->loggerMock);
+
+        $event = new BuildSuggestionCollectionEvent($query);
+
+        $this->loggerMock
+            ->expects($this->once())
+            ->method('error');
+
+        $subscriber->onBuildSuggestionCollectionEvent($event);
+    }
+
+    private function performTestOnContentSuggestion(): void
     {
         $query = new SuggestionQuery('test', 10, 'eng-GB');
         $searchService = $this->getSearchServiceMock();
         $mapper = $this->getSearchHitToContentSuggestionMapperMock();
 
-        $subscriber = new ContentSuggestionSubscriber($searchService, $mapper);
+        $sortingDefinitionRegistry = $this->getSortingDefinitionRegistryMock();
+
+        $subscriber = new ContentSuggestionSubscriber($searchService, $mapper, $sortingDefinitionRegistry);
```

Comment:
You can, although it is slightly out of scope as it enhances the existing test, make assertions about logger not being called (`self::never()`).